README.md

```diff
@@ -0,0 +1,101 @@
+# Operator-custom metrics
+This library assists in collecting operator custom-metrics from operators and exposing them to prometheus instance.
+
+## Prerequisites
+In order to use the functions provided by this library download the package inside the `vendor` directory of your operator code.
+The following command can be used:
+
+```bash
+go get github.com/openshift/operator-custom-metrics
+```
+
+## Using operator-custom metrics library
+
+### Description
+The library is designed to register the metrics provided by the user to prometheus. The user can provide the **metrics configuration** based on which the metrics are collected.
+The following functions are performed by the library:
+1. Register custom metrics provided by the user to prometheus.
+2. Publish metrics at the endpoint and path based on the metrics configuration.
+3. Create a service resource, which selects the pods deployed by the operator based on the labels.
```

Comment:
Don't worry about defining what a service is unless the user needs to provide information to set up these labels.